Sources/CartonKit/Server/Server.swift

```diff
@@ -227,9 +227,26 @@ public actor Server {
   }
 
   /// Blocking function that starts the HTTP server.
-  public func run() throws {
-    defer { app.shutdown() }
-    try app.run()
+  public nonisolated func run() async throws {
+    // Explicitly hop to another thread to avoid blocking the thread that is running the actor executor
+    let _: () = try await withCheckedThrowingContinuation { continuation in
```

Comment:
Can this be…
```suggestion
    try await withCheckedThrowingContinuation { continuation in
```

…since the `Void` value is not used?